src/dstack/api/_public/runs.py

```diff
@@ -691,28 +691,31 @@ def exec_plan(
         logger.warning("The exec_plan() method is deprecated in favor of apply_plan().")
         return self.apply_plan(run_plan=run_plan, repo=repo, reserve_ports=reserve_ports)
 
-    def list(self, all: bool = False) -> List[Run]:
+    def list(self, all: bool = False, limit: Optional[int] = None) -> List[Run]:
         """
         List runs.
 
         Args:
             all: Show all runs (active and finished) if `True`.
+            limit: Limit the number of runs to return. Must be less than 100.
 
         Returns:
             List of runs.
         """
         # Return only one page of latest runs (<=100). Returning all the pages may be costly.
         # TODO: Consider introducing `since` filter with a reasonable default.
-        only_active = not all
+        only_active = not all and limit is None
         runs = self._api_client.runs.list(
             project_name=self._project,
             repo_id=None,
             only_active=only_active,
+            limit=limit or 100,
         )
         if only_active and len(runs) == 0:
             runs = self._api_client.runs.list(
                 project_name=self._project,
                 repo_id=None,
+                limit=limit or 100,
             )[:1]
```

Comment:
(nit) Since we only need the last run here, we can always pass `limit=1`. This may speed up the response a bit
```suggestion
            runs = self._api_client.runs.list(
                project_name=self._project,
                repo_id=None,
                limit=1,
            )
```

Reply:
@jvstme Active runs can be many

Reply:
This block is used for loading one stopped run if there are no active runs. It already sets the limit to one run, expect it does so after receiving the response - `[:1]`